swarm/api/routes/evolution.py

```
@@ -430,6 +454,14 @@ async def apply_evolution_patch_endpoint(
                 },
             )
 
+    try:
+        if run_id is not None:
+            validate_path_component(run_id, "run_id")
+        if patch_id is not None:
+            validate_path_component(patch_id, "patch_id")
+    except ValueError as e:
+        raise HTTPException(status_code=400, detail={"error": "invalid_path", "message": str(e)})
```

Ruff (0.15.5)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)